src/EGraphs/uniquequeue.jl

```diff
@@ -1,8 +1,27 @@
 """
 A data structure to maintain a queue of unique elements.
 Notably, insert/pop operations have O(1) expected amortized runtime complexity.
-"""
 
+Note: this is really more of a stack than a queue, because it is LIFO (last in
+first out) rather than FIFO (first in, last out). That is,
+
+```julia
+q = UniqueQueue{Int}()
+push!(q, 1)
+push!(q, 2)
+assert(pop!(q) == 2)
+assert(pop!(q) == 1)
+```
```

Comment:
Nice catch. This is modeled after egg's unique queue. We should check `egg`'s source code and see if it's FIFO or LIFO. Could mean a big performance gain if they're going FIFO and we're doing LIFO, we should try to align with their implementation.

Reply:
It does seem like they are going FIFO:

https://github.com/egraphs-good/egg/blob/b3a53e9e575dc67ce1e8320e4e488f2873a67482/src/util.rs#L152-171

According to https://doc.rust-lang.org/std/collections/struct.VecDeque.html, push_back and pop_front is FIFO, and also VecDeque is implemented with a ring buffer.

Reply:
Ah, we should fix then